src/main/java/com/iridium/iridiumskyblock/gui/BlockLimitGUI.java

```diff
@@ -0,0 +1,86 @@
+package com.iridium.iridiumskyblock.gui;
+
+import com.iridium.iridiumcore.dependencies.xseries.XMaterial;
+import com.iridium.iridiumcore.utils.InventoryUtils;
+import com.iridium.iridiumcore.utils.ItemStackUtils;
+import com.iridium.iridiumcore.utils.StringUtils;
+import com.iridium.iridiumskyblock.IridiumSkyblock;
+import com.iridium.iridiumskyblock.upgrades.BlockLimitUpgrade.LimitedBlock;
+import org.bukkit.entity.Player;
+import org.bukkit.event.inventory.InventoryClickEvent;
+import org.bukkit.inventory.Inventory;
+import org.bukkit.inventory.ItemStack;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+/**
+ * GUI which shows block limits.
+ *
+ * @see LimitedBlock
+ */
+public class BlockLimitGUI extends GUI {
+
+    private final int level;
+    private int page = 1;
+
+    /**
+     * The default constructor.
+     */
+    public BlockLimitGUI(int level, Inventory previousInventory) {
+        super(IridiumSkyblock.getInstance().getInventories().blockLimit, previousInventory);
+        this.level = level;
+    }
+
+    @Override
+    public void addContent(Inventory inventory) {
+        inventory.clear();
+
+        InventoryUtils.fillInventory(inventory, getNoItemGUI().background);
+
+        inventory.setItem(inventory.getSize() - 3, ItemStackUtils.makeItem(IridiumSkyblock.getInstance().getInventories().nextPage));
+        inventory.setItem(inventory.getSize() - 7, ItemStackUtils.makeItem(IridiumSkyblock.getInstance().getInventories().previousPage));
+
+        IridiumSkyblock.getInstance().getUpgrades().blockLimitUpgrade.upgrades.get(level).limits.entrySet().stream()
+                .filter(limitedBlockEntry -> (page == 1 ? limitedBlockEntry.getValue().page <= 1 : limitedBlockEntry.getValue().page == page))
+                .forEachOrdered(limitedBlock -> {
+                    XMaterial material = limitedBlock.getKey();
+                    LimitedBlock limitInfo = limitedBlock.getValue();
+                    ItemStack blockItem = ItemStackUtils.makeItem(material, 1, StringUtils.color(limitInfo.name), getColoredValueLore(limitInfo.value));
```

Comment:
This is kinda weird, should be using
ItemStackUtils#makeItem(Item, Collection<Placeholder>)

Reply:
What do you mean? I did it almost the same way as here
https://github.com/Iridium-Development/IridiumSkyblock/blob/fcd68a3cec65569074ae68498659ba0a11f5af0b/src/main/java/com/iridium/iridiumskyblock/gui/BlockValueGUI.java#L51-L54